components/Search.tsx

```diff
@@ -0,0 +1,353 @@
+        }
+        creators {
+          id
+          name
+          givenName
+          familyName
+        }
+        descriptions {
+          description
+          descriptionType
+        }
+        publicationYear
+        publisher
+        version
+        citationCount
+        viewCount
+        downloadCount
+      }
+    }
+  }
+`
+
+export const Search: React.FunctionComponent<Props> = () => {
+  const router = useRouter()
+  // set initial searchQuery to query parameter in route
+  const [searchQuery, setSearchQuery] = React.useState(router.query.query as string || "")
+  const [searchResults, setSearchResults] = React.useState([])
+  const { loading, error, data, refetch, fetchMore } = useQuery<ContentQueryData, ContentQueryVar>(
+    CONTENT_GQL,
+    {
+      errorPolicy: 'all',
+      variables: { query: "", cursor: "", published: router.query.published as string, resourceTypeId: router.query['resource-type'] as string }
+    }
+  )
+
+  const onSearchChange = (e: React.FormEvent<HTMLInputElement>): void => {
+    // sync searchQuery and query parameter in route
+    router.push('/?query=' + e.currentTarget.value)
+    setSearchQuery(e.currentTarget.value)
+  }
+
+  const onSearchClear = () => {
+    // reset searchQuery and sync with query parameter in route
+    router.push('/')
+    setSearchQuery('')
+  }
+
+  const onNextPage = () => {
+    loadMore(data.works.pageInfo.endCursor)
+  }
+
+  const loadMore = (cursor: String) => {
+    fetchMore(
+      { variables: { cursor: cursor },
+      updateQuery: (previousResult: ContentQueryData, { fetchMoreResult }) => {
+        if (!fetchMoreResult) { return previousResult }
+
+        const newNodes = fetchMoreResult.works.nodes
+        const pageInfo = fetchMoreResult.works.pageInfo
+        const totalCount = fetchMoreResult.works.totalCount
+        const published = fetchMoreResult.works.published
+        const resourceTypes = fetchMoreResult.works.resourceTypes
+        const registrationAgencies = fetchMoreResult.works.registrationAgencies
+
+        return newNodes.length
+          ? {
+              works: {
+                __typename: previousResult.works.__typename,
+                nodes: [...previousResult.works.nodes, ...newNodes],
+                pageInfo,
+                totalCount,
+                published,
+                resourceTypes,
+                registrationAgencies
+              }
+          }
+          : previousResult
+        }
+      })
+  }
+
+  React.useEffect(() => {
+    const typingDelay = setTimeout(() => {
+      try {
+        // only trigger search with at least one character as input
+        // otherwise reset search results
+        if (searchQuery.length > 0) {
+          refetch({ query: searchQuery, cursor: "", published: router.query.published as string, resourceTypeId: router.query['resource-type'] as string})
+        } else {
+          setSearchResults([])
+        }
+      } catch(e) {
+        console.log(e)
```

Comment:
We probably should consider in grander scope how we want to do logging.

Reply:
Agree. This is (mostly) a server application.